src/GridInterpolations.jl

```diff
@@ -157,12 +157,12 @@ function interpolants(grid::RectangleGrid, x::AbstractVector)
     cut_counts = grid.cut_counts
     cuts = grid.cuts
 
-
     # Reset the values in index and weight:
-    index = @MVector(ones(Int, 2^dimensions(grid)))
-    index2 = @MVector(ones(Int, 2^dimensions(grid)))
-    weight = @MVector(zeros(eltype(x), 2^dimensions(grid)))
-    weight2 = @MVector(zeros(eltype(x), 2^dimensions(grid)))
+    num_points = 2^dimensions(grid)
+    index = MVector{num_points, Int}(undef)
+    index2 = MVector{num_points, Int}(undef)
+    weight = MVector{num_points, eltype(x)}(undef)
+    weight2 = MVector{num_points, eltype(x)}(undef)
```

Comment:
I am a bit concerned about this change. How can we be completely sure that it does not change the code logic to initialize these to `undef` rather than ones or zeros? (note that the tests may pass most of the time even if the code relies on zeros since memory is often initialized to zeros or small numbers.)

I think we should either leave this as is (with ones or zeros) or carefully go through the code below looking at all of the assignments to these variables and write out a brief proof that the initialization does not matter.

Reply:
Good point. Zeroing memory typically does not have an impact on runtime.

Reply:
The original goal for that change was to reduce allocations, because the `ones()` and `zeros()` calls would theoretically be allocating arrays, although I guess the compiler could be factoring that out. From looking at the code I don't believe the arrays need to be explicitly initialized, but just in case I've added it back in, just using broadcasting instead of allocating new arrays for the initialization.

Reply:
Thanks @ctessum ! Yeah, If I remember correctly, we verified that `@MVector zeros(...)` etc. does not allocate as long as the MVector does not leave the scope of the function. I added an explanatory note.

If we wanted to avoid this and be explicit about proving that the initialization doesn't matter, we could use `PushVector`s backed with `MVector`s, but that might require some work.